### PR TITLE
Add support for AMD Phoenix APUs with Radeon 740M/760M/780M/880M/890M

### PR DESCRIPTION
**Pull request by akohlmey**
This pull request aims to add support for the AMD GFX1103 architecture which is used in AMD "Phoenix" APUs, i.e. CPU/GPU combo chips used in notebooks and mini-pcs. The GPU would be listed as Radeon 740M, 760M, 780M, 880M, or 890M. 

This was tested with an `AMD Ryzen 7 7840HS w/ Radeon 780M Graphics` on Fedora 40 with bundled ROCm 6.1.2. using the LAMMPS development branch.

This mostly follows a previous pull request adding support for GFX1100 and is mostly guesswork based on similar entries.

Please let me know if this is missing something important. I'm leaving the pull request "open" so that Kokkos maintainers can apply changes directly.

@stanmoore1 if this can go in, we should port this into the Kokkos version bundled with LAMMPS for the next release.

### CODE REVIEW

**Comment by Rombur** [comment]
According to AMD's [documentation](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html), it is not officially supported but I am fine accepting this PR.

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -1044,6 +1044,12 @@ FOREACH(ARCH IN LISTS SUPPORTED_AMD_ARCHS)
       SET(KOKKOS_ARCH_NAVI1100 ON)
       BREAK()
     ENDIF()
+    STRING(REGEX MATCH "1103" IS_1103 ${ARCH})
+    IF(IS_1103)
+      SET(KOKKOS_ARCH_AMD_GFX1103 ON)
+      SET(KOKKOS_ARCH_NAVI ON)
+      BREAK()
+    ENDIF()
```
Comment (Rombur) [inline cmake/kokkos_arch.cmake:1052]: This was done for backward compatibility. This is not necessary when adding a new backend.

**Review by Rombur — CHANGES_REQUESTED** [review]

File: cmake/kokkos_arch.cmake (hunk 1 above)
Comment (akohlmey) [reply cmake/kokkos_arch.cmake]: it is gone now.

**Comment by akohlmey** [comment]
> According to AMD's [documentation](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html), it is not officially supported but I am fine accepting this PR.

Perhaps not yet.

File: Makefile.kokkos [hunk 2]
```diff
@@ -455,6 +455,7 @@ KOKKOS_INTERNAL_USE_ARCH_AMD_GFX1100 := $(call kokkos_has_string,$(KOKKOS_ARCH),
 ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX1100), 0)
   KOKKOS_INTERNAL_USE_ARCH_AMD_GFX1100 := $(call kokkos_has_string,$(KOKKOS_ARCH),NAVI1100)
 endif
+KOKKOS_INTERNAL_USE_ARCH_AMD_GFX1100 := $(call kokkos_has_string,$(KOKKOS_ARCH),AMD_GFX1103)
```
Comment (dalg24) [inline Makefile.kokkos:458]:
```suggestion
KOKKOS_INTERNAL_USE_ARCH_AMD_GFX1103 := $(call kokkos_has_string,$(KOKKOS_ARCH),AMD_GFX1103)
```

**Review by Rombur — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Ignoring SYCL failure on Jenkins

**Pull request merged**